app/assets/javascripts/preferences/panes/account/ChangeEmail.tsx

```diff
@@ -0,0 +1,80 @@
+import ModalDialog, {
+  ModalDialogButtons,
+  ModalDialogDescription,
+  ModalDialogLabel
+} from '@/components/shared/ModalDialog';
+import { FunctionalComponent } from 'preact';
+import { DecoratedInput } from '@/components/DecoratedInput';
+import { Button } from '@/components/Button';
+import { useState } from 'preact/hooks';
+import { SNAlertService } from '@node_modules/@standardnotes/snjs';
+import { EMAIL_REGEX } from '@Views/constants';
+import { HtmlInputTypes } from '@/enums';
+
+type Props = {
+  onCloseDialog: () => void;
+  snAlert: SNAlertService['alert']
+};
+
+const ChangeEmail: FunctionalComponent<Props> = ({ onCloseDialog, snAlert }) => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+
+  const handleSubmit = () => {
+    let errorMessage = '';
+    if (email.trim() === '' || password.trim() === '') {
+      errorMessage = 'Please populate all fields';
+    } else if (!EMAIL_REGEX.test(email)) {
```

Comment:
What do you think about moving this into a util function in utils.ts? Probably lots of places we'll want to validate email.

Reply:
Good point, done.